lib/Dialect/AIEVec/Transforms/VectorToVectorConversions.cpp

```diff
@@ -300,14 +444,21 @@ static void configureAIEMLCanonicalizeLegalizations(ConversionTarget &target) {
       [](vector::TransferReadOp op) {
         return !op.getPermutationMap().isConstant() &&
                getTransferReadAlignmentOffset(op, op.getVectorType(), 256)
-                       .value_or(0) == 0;
+                       .value_or(0) == 0 &&
+               op.getVector().getType().getRank() < 2;
+      });
+  target.addDynamicallyLegalOp<vector::TransferWriteOp>(
+      [](vector::TransferWriteOp op) {
+        return cast<VectorType>(op.getVector().getType()).getRank() < 2;
       });
 }
 
 static void
 populateAIEMLCanonicalizeConversionPatterns(RewritePatternSet &patterns) {
   patterns.add<SplitUnalignedTransferReadPattern>(patterns.getContext(), 1024,
                                                   256);
+  patterns.add<FlattenMultDimTransferReadPattern,
+               FlattenMultDimTransferWritePattern>(patterns.getContext());
```

Comment:
Somehow this broke some of my E2E stuff but in the strangest way - on linux it seems to be no problem but on mac there's an issue...

Reply:
ugh false alarm - some stale link/compilation units somewhere